### PR TITLE
Add `appearance` prop to `ic-breadcrumb` & `ic-breadcrumb-group`

### PR DESCRIPTION
**Pull request by GCHQ-Developer-112**
## Summary of the changes
Add default, dark and light variants of `ic-breadcrumb` and `ic-breadcrumb-group` through `appearance` prop.

## Related issue
#1259 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

File: packages/react/src/stories/ic-breadcrumb-group.stories.mdx [hunk 1]
```diff
@@ -88,6 +88,146 @@ import {
   </Story>
 </Canvas>
 
+### Appearance
+
+<Canvas>
+  <Story name="Appearance">
+    <IcBreadcrumbGroup appearance="default">
```
Comment (jd239) [inline packages/react/src/stories/ic-breadcrumb-group.stories.mdx:95]: Is the `default` value required as it is set to default by default in the code.

File: packages/web-components/src/components/ic-breadcrumb-group/ic-breadcrumb-group.tsx [hunk 2]
```diff
@@ -42,6 +48,12 @@ export class BreadcrumbGroup {
       this.el.querySelectorAll(this.IC_BREADCRUMB)
     );
 
+    if (this.appearance != "default") {
```
Comment (jd239) [inline packages/web-components/src/components/ic-breadcrumb-group/ic-breadcrumb-group.tsx:51]:
```suggestion
    if (this.appearance !== "default") {
```

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.tsx
Comment (jd239) [inline packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.tsx:1]: Do we need to add a story where we leave the appearance on the breadcrumbGroup as default but change the appearance on the individual breadcrumb?

**Review by jd239 — CHANGES_REQUESTED** [review]
Nice update. Just a few comments

File: packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.tsx
Comment (GCHQ-Developer-112) [reply packages/web-components/src/components/ic-breadcrumb/ic-breadcrumb.tsx:1]: Based on the designs, I don't think we should allow this functionality. The designs show that the breadcrumbs that are always unified in colour despite the variant you choose. I've made the appearance prop on `ic-breadcrumb` internal as a result

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by jd239 — APPROVED** [review]

**Pull request merged**